### DISCUSSION TITLE
GITHUB_ENV. Invalid environment variable format

### DISCUSSION THREAD

**Discussion by AndreiPaulau** [community/community#24917] [Actions]
<p>I’ve stucked with an error and believe tried everything (different quotes, sed, escape characters)<br>
Error: Invalid environment variable format ‘“<a href="http://some.site.net" rel="noopener nofollow ugc">some.site.net</a>”’</p>
<p>Here is the following step:</p>
<ul>
<li>name: Set URL for deployment<br>
run: |<br>
echo “CF_URL=$(terraform output cf_url)” &gt;&gt; $GITHUB_ENV</li>
</ul>
<p>Run echo “CF_URL=$(terraform output cf_url)” &gt;&gt; $GITHUB_ENV</p>
<p>Error: Unable to process file command ‘env’ successfully.<br>
Error: Invalid environment variable format ‘“<a href="http://some.site.net" rel="noopener nofollow ugc">some.site.net</a>”’</p>
<p>I think this is something stupid regrading dots in env var. Any clue?<br>
Much appreciated</p>

**Comment by airtower-luna** [comment]
<p>Dots in an environment variable shouldn’t be an issue. Could the output of the <code>terraform</code> command possibly contain multiple lines? In that case you’d need to use a slightly different format, see <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#multiline-strings">Multiline strings</a>.</p>

**Comment by AndreiPaulau** [comment]
<p>Thanks, your assumption was right! <a href="https://github.community/u/airtower-luna">airtower-luna</a><br>
I setup terraform through the action (hashicorp/setup-terraform@v1). The output was multiline (BTW why?)</p>
<p>"[command]/home/runner/work/_temp/adgadfga/terraform-bin output cf_url<br>
/home/runner/work/_temp/adgadfga/terraform-bin output cf_url<br>
<a href="http://some.site.net" class="onebox" target="_blank" rel="noopener nofollow ugc">some.site.net</a></p>
<p>Will try to trim it if possible. May be someone already faced such behaviour?</p>

**Comment by AndreiPaulau** [comment, marked as answer]
<p>Finally I was able to solve this by cleaning the output out of noise:<br>
run: |<br>
echo ‘CF_URL&lt;&lt;EOF’ &gt;&gt; $GITHUB_ENV<br>
terraform output cf_url | grep .net | grep -v “::” | sed -e ‘s/"//g’ &gt;&gt; $GITHUB_ENV<br>
echo ‘EOF’ &gt;&gt; $GITHUB_ENV<br>
Still need to read regarding GITHUB_ENV</p>
